doc/source/overview/dev_practices.rst

```diff
@@ -194,4 +194,372 @@ should not wait until a minor release. These are the steps for a patch release:
+should be tested, but the tests written will depend on purpose of the library.
+For example, a library that is wrapping a gRPC interface would include tests of
+the gRPC methods exposed by the proto files and wrapped by the Python library, but would not be expected to test the functionality of the server.
+
+For example, if testing the gRPC method ``GetNode``:
+
+.. code::
+
+   message Node
+   {
+     int32      id = 1;
+     double     x = 2;
+     double     y = 3;
+     double	z = 4;
+   }
+
+   message NodeRequest {
+     int32		num = 1;
+   }
+
+   message NodeResponse {
+     Node		node = 1;
+   }
+
+  service SomeService{
+
+     rpc GetNode(NodeRequest)				returns (NodeResponse);
+     // other methods
+   }
+
+Then youyour unit test would test the wrapped python function (for example,
+``get_node``).  For example the ``get_node`` method might be implemented as:
+
+.. code:: python
+
+   from ansys.product.service.v0 import service_pb2
+
+   def get_node(self, index):
+       """Return the coordinates of a node for a given index.
+
+       Parameters
+       ----------
+       index : int
+           Index of the node.
+
+       Returns
+       -------
+       tuple
+           Coordinates of the node.
+
+       Examples
+       --------
+       >>> from ansys.product.service import SomeService
+       >>> srv = SomeService()
+       >>> srv.create_node(1, 4.5, 9.0, 3.2)
+       >>> node = srv.get_node(1)
+       >>> node
+       (4.5, 9.0, 3.2)
+
+       """
+       resp = service_pb2.GetNode(index=index)
+       return resp.x, resp.y, resp.z
+
+Your test would be implemented (within ``tests/test_nodes.py``):
+
+.. code::
+
+   def test_get_node(srv):
+       srv.clear()
+
+       node_index = 1
+       node_coord = 0, 10, 20
+       srv.create_node(node_index, node_coord*)
+       assert srv.get_node(node_index) == node_coord
+
+The goal of the unit test should be to test the wrapping of the interface rather
+than the product or service itself. In the case of the ``GetNode``, this method
+should have already been tested when designing and developing the service.
+
+
+Testing Framework
+~~~~~~~~~~~~~~~~~
+For consistency, PyAnsys tools and libraries should use either the `unit test
+<some link>`_ or `pytest <some link>`_ frameworks for unit testing. As detailed
+in the :ref:`repo_dir_struct`, unit tests should be placed into the ``tests``
+directory in the of the root directory of the library::
+
+   tests/
+       test_basic.py
+       test_advanced.py
+
+Furthermore, any testing dependencies requirements should be included when using ``setup.py`` within a ``requirements_tests.txt`` file that is installed via::
+
+.. code::
+
+   pip install -r requirements_tests.txt
+
+Or included in ``pyproject.toml``. For example, when using the `filt
+<missing_link>`_ build system::
+
+   [project.optional-dependencies]
+   test = [
+       "pytest>=2.7.3",
+       "pytest-cov",
+   ]
+
+And then installed via::
```

Comment:
We should format this as a complete sentence. Are the dependencies installed via the code presented next?

Reply:
They are installed with this code snippet but it is done in a separate command prompt.
The code above is part of a file used for packaging.
I think that we can let it this way to show the separation.